src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import { postNewComment } from "../utils/api";
+import { UserContext } from "../contexts/User";
+import { useContext } from "react";
+
+const CommentAdder = ({ review_id, setComments }) => {
+  const { user } = useContext(UserContext);
+  const [newComment, setNewComment] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    postNewComment(review_id, user.username, newComment).then(({ comment }) => {
+      setComments((currComments) => {
```

Comment:
You should be doing this setComments before the API call - this is what we mean by optimistic rendering. The setComments is synchronous but you're still waiting for the asynchronous API call.